### ISSUE TITLE
Remove or replace Nuxt reference and logic for Vue3 migration

### ISSUE THREAD

**Issue by cnotv** [rancher/dashboard#10491]
# Description

Some lingering code inherited form Nuxt is going to clash with the breaking changes from the Vue3 migrations.
These needs to be either corrected to match the syntax, removed if unnecessary or replaced with a new implementation.

## Context

Given [the deprecation of the emitters](https://v3-migration.vuejs.org/breaking-changes/events-api.html#event-bus), all these cases have to be handled as they will break for sure:
- [x] `$nuxt.$on`
- [x] `$nuxt.$off`
- [x] `$nuxt.$emit`

Other breaking cases:
- [x] `$nuxt.$nextTick` ([docs](https://v3-migration.vuejs.org/breaking-changes/global-api-treeshaking.html#global-api-treeshaking))
- [x] `Vue.util.defineReactive`, used in more places
- [x] `Vue.__nuxt__fetch__mixin__`
- [x] ~`Component = Vue.extend(Component);` ([docs](https://v3-migration.vuejs.org/breaking-changes/global-api.html))~ Can be removed in migration branch
- [x] `Vue[installKey]` ([lines](https://github.com/rancher/dashboard/blob/master/shell/initialize/index.js#L165-L178))
- [x] ~`vueDirective.apply(Vue, arguments);`  ([line](https://github.com/rancher/dashboard/blob/master/shell/initialize/index.js#L57))~ Can be removed in migration branch
- [x] `Vue.config.$nuxt.$nuxt = true;` and `Vue.config().$nuxt = {};` ([line](https://github.com/rancher/dashboard/blob/master/shell/initialize/client.js#L112-L114))
- [x] `vueApp.config().$nuxt`  ([line](https://github.com/rancher/dashboard/blob/master/shell/initialize/client.js#L73))
- [x] ~`nuxt.publicRuntimeConfig`, this is needed to start the app ([line](https://github.com/rancher/dashboard/blob/master/shell/initialize/client.js#L120))~ Seems can be left untouched

Optional other cases which would be nice to get rid of:
- [x] `$nuxt.$nuxt`
- [x] `$nuxt.$store`
- [x] `$nuxt.$route`
- [x] `$nuxt.$router`

**Comment by codyrancher** [comment]
What's the issue you're seeing with `Vue.util.defineReactive`? By any chance did it just get moved? I found it in the vue code base https://github.com/vuejs/vue/blob/main/src/core/observer/index.ts#L128.

**Comment by codyrancher** [comment]
@cnotv here's the current progress https://github.com/codyrancher/dashboard/pull/3.

I'll continue on the list.

**Comment by cnotv** [comment]
> What's the issue you're seeing with `Vue.util.defineReactive`? By any chance did it just get moved? I found it in the vue code base https://github.com/vuejs/vue/blob/main/src/core/observer/index.ts#L128.

In Vue 3 you cannot import anymore Vue from the library, there's just the function `createApp`.

**Comment by cnotv** [comment]
@codyrancher found another case which I forgot: https://github.com/rancher/dashboard/blob/master/shell/initialize/index.js#L165-L178

**Comment by cnotv** [comment]
I'm adding other cases while crossing them because the app breaks, just in case we miss them.

**Comment by codyrancher** [comment]
@cnotv I've opened a PR which takes care of each of the checked items on the list. For the remaining items:

1. `Component = Vue.extend(Component);` I think we could remove this in your branch once we have all components/pages defined in the same way but not something we can push in master right now
2. `vueDirective.apply(Vue, arguments);` We already talked about this. I think this can be removed in your branch since the way directives are defined and installed is different in vue 3. I don't think we need to make this extra check.
3. `nuxt.publicRuntimeConfig` What's the problem with this one? This is really just a string replacement from vue.config.js
